core/cthreads.c

```diff
@@ -0,0 +1,291 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "cthreads.h"
+
+#if __unix__
+#include <pthread.h>
+void *__cthreads_pthread_function_wrapper(void *data) {
+  struct cthreads_args *args = data;
+  args->func(args->data);
+
+  free(data);
+
+  return NULL;
+}
+#elif _WIN32
+#include <windows.h>
+DWORD WINAPI __cthreads_winthreads_function_wrapper(void *data) {
+  struct thread_args *args = data;
+  args.func(args.data);
+
+  free(data);
+
+  return TRUE;
+}
+#endif
+
+int cthreads_thread_create(struct cthreads_thread *thread, void *(*func)(void *data), void *data) {
+  #ifdef __unix__
+    pthread_t pthread;
+
+    struct cthreads_args *args = malloc(sizeof(struct cthreads_args));
+    args->func = func;
+    args->data = data;
+
+    int res = pthread_create(&pthread, NULL, __cthreads_pthread_function_wrapper, (void *)args);
+
+    thread->pThread = pthread;
+
+    return res;
+  #elif _WIN32
+    struct cthreads_args *args = malloc(sizeof(struct cthreads_args));
+    args->func = func;
+    args->data = data;
+
+    HANDLE thread = CreateThread(NULL, 0, __cthreads_winthreads_function_wrapper, args, 0, NULL);
+
+    threading->wThread = thread;
+
+    return 0;
+  #endif
+}
+
+int cthreads_thread_detach(struct cthreads_thread *thread) {
+  #ifdef __unix__
+    return pthread_detach(thread->pThread);
+  #elif _WIN32
+    return CloseHandle(thread->wThread);
+  #endif
+}
+
+void cthreads_thread_close(void *code) {
+  #ifdef __unix__
+    pthread_exit(code);
+  #elif _WIN32
+    ExitThread((DWORD)code);
+  #endif
+
+  return;
+}
+
+struct cthreads_mutex __cthreads_mutex_init() {
+  struct cthreads_mutex mutex;
+
+  #ifdef __unix__
+    pthread_mutex_t pMutex = PTHREAD_MUTEX_INITIALIZER;
+    pthread_mutex_init(&pMutex, NULL);
+
+    mutex.pMutex = pMutex;
```

Comment:
I don't think you should be initializing as such.. PTHREAD_MUTEX_INITIALIZER followed by a `pthread_mutex_init()` is redundant!
```suggestion
    pthread_mutex_init(&mutex.pMutex, NULL);
```